### PR TITLE
Feature/jhbiggs/1581 remove dagger android

### PR DESCRIPTION
**Pull request by jhbiggs**
#1581 Successfully extricated dagger-android but now it won't pass githooks.

### CODE REVIEW

**Comment by jhbiggs** [comment]
Ah, I see the problems in the error output.  Working on the errors...

File: app/src/main/java/org/kiwix/kiwixmobile/di/components/ServiceComponent.kt [hunk 1]
```diff
@@ -35,6 +36,8 @@ interface ServiceComponent {
 
     @BindsInstance fun service(service: Service): Builder
 
+    @BindsInstance fun notificationManager(notificationManager: NotificationManager): Builder
```
Comment (macgills) [inline app/src/main/java/org/kiwix/kiwixmobile/di/components/ServiceComponent.kt]: you don't need to do this, in our applicationModule
```
  @Provides
  @Singleton
  internal fun provideNotificationManager(context: Context): NotificationManager =
    context.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
```
We haven't got into the specifics of why DI Is especially complicated in a multimodule project but in CoreComponent we expose this dependency across modules like so
```
  fun notificationManager(): NotificationManager
```

Using `@BindsInstance` is saying "to create this component I _need_ these objects"

File: core/build.gradle.kts [hunk 2]
```diff
@@ -58,7 +58,7 @@ dependencies {
   }
 
   // Leak canary
-  debugImplementation(Libs.leakcanary_android)
+  implementation(Libs.leakcanary_android)
```
Comment (macgills) [inline core/build.gradle.kts]: https://square.github.io/leakcanary/getting_started/ leakcanary expects debugImlementation

File: core/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -66,7 +66,7 @@
   <string name="pref_credits_title">Credits</string>
   <string name="tts_lang_not_supported">The language of this page is not supported, or appropriate language data was not installed. The article may not be properly read.</string>
   <string name="no_reader_application_installed">Could not find an installed application for this type of file</string>
-  <string name="custom_app_missing_content">Your application is corrupted.\nThis might happen when you remove files on the SD Card.\nYou need to uninstall then reinstall the App from the Play Store.</string>
+  <string name="custom_app_missing_content">Your application is corrupted.\nThis might happen when you remove files on the SD Card.\nYou need to uninstall then reinstall the App from the Play Store.</string>
```
Comment (macgills) [inline core/src/main/res/values/strings.xml]: I am not sure why that space is different to a regular space and not too bothered checking but revert this change

File: gradle.properties [hunk 4]
```diff
@@ -1,6 +1,18 @@
...
+# Default value: -Xmx1024m -XX:MaxPermSize=256m
+# org.gradle.jvmargs=-Xmx2048m -XX:MaxPermSize=512m -XX:+HeapDumpOnOutOfMemoryError -Dfile.encoding=UTF-8
+#
+# When configured, Gradle will run in incubating parallel mode.
+# This option should only be used with decoupled projects. More details, visit
+# http://www.gradle.org/docs/current/userguide/multi_project_builds.html#sec:decoupled_projects
+# org.gradle.parallel=true
+#Sat Jan 11 17:55:48 CST 2020
 android.enableJetifier=true
+org.gradle.jvmargs=-Xmx2048M -Dkotlin.daemon.jvm.options\="-Xmx2048M"
```
Comment (macgills) [inline gradle.properties]: were you having problems with our jvmargs? regardless these should not be changed as part of this ticket

File: core/src/main/java/org/kiwix/kiwixmobile/core/base/BaseActivity.kt [hunk 5]
```diff
@@ -36,16 +37,14 @@ abstract class BaseActivity : AppCompatActivity() {
 
   private var unbinder: Unbinder? = null
 
+  protected abstract fun injection(coreComponent: CoreComponent)
+
   override fun onCreate(savedInstanceState: Bundle?) {
-    injection()
+    injection(CoreApp.getCoreComponent())
```
Comment (macgills) [inline core/src/main/java/org/kiwix/kiwixmobile/core/base/BaseActivity.kt]: when you were adding this did you rewrite the function first? There is a couple ways to do this so the change propagates to all the call sites

just add it to get an error
image
and we want to add a parameter
image
that gives us the ChangeSignature modal and we just have to hit the refactor button (or alt+r) and all existing callsites are modified
as I can see in my list of changed files
image

Alternatively you can just go straight to the ChangeSignature modal by right clicking on a method>Refactor>change signature, or placing your cursor on a method and pressing ctrl+f6
